tools/alpine/Makefile

```diff
@@ -2,22 +2,34 @@
 
 ORG?=linuxkit
 IMAGE=alpine
+
+DOCKER_CONTENT_VAR=1
+PKG_DEPS=packages
+ifeq ($(shell uname -m), x86_64)
 BASE=alpine:3.6
+PKG_DEPS += packages.x86_64
+endif
+
+ifeq ($(shell uname -m), aarch64)
+BASE=arm64v8/alpine:3.6
+PKG_DEPS += packages.aarch64
+DOCKER_CONTENT_VAR=0
+endif
 
 default: push
 
 show-tag:
 	@sed -n -e '1s/# \(.*\/.*:[0-9a-f]\{40\}\)/\1/p;q' versions
 
-hash: Dockerfile Makefile packages
-	DOCKER_CONTENT_TRUST=1 docker pull $(BASE)
-	docker build --no-cache -t $(IMAGE):build .
+hash: Dockerfile Makefile $(PKG_DEPS)
+	DOCKER_CONTENT_TRUST=$(DOCKER_CONTENT_VAR) docker pull $(BASE)
+	docker build --no-cache --build-arg BASE=$(BASE) -t $(IMAGE):build .
 	docker run --rm $(IMAGE):build sh -c 'echo Dockerfile /lib/apk/db/installed $$(find /mirror -name '*.apk' -type f) $$(find /go/bin -type f) | xargs cat | sha1sum' | sed 's/ .*//' > $@
 
 push: hash
-	DOCKER_CONTENT_TRUST=1 docker pull $(ORG)/$(IMAGE):$(shell cat hash) || \
+	DOCKER_CONTENT_TRUST=$(DOCKER_CONTENT_VAR) docker pull $(ORG)/$(IMAGE):$(shell cat hash) || \
 		(docker tag $(IMAGE):build $(ORG)/$(IMAGE):$(shell cat hash) && \
-		 DOCKER_CONTENT_TRUST=1 docker push $(ORG)/$(IMAGE):$(shell cat hash))
+		 DOCKER_CONTENT_TRUST=$(DOCKER_CONTENT_VAR) docker push $(ORG)/$(IMAGE):$(shell cat hash))
 	echo "# $(ORG)/$(IMAGE):$(shell cat hash)" > versions
```

Comment:
This file is checked in and is expected to contain the exact versions present in the image hash in the header. Given that this package is built with network access it might be quite hard to ensure that there is no skew due to arches being built at different times (or even atthe same time if there is skew in the upstream alpine repo).

IMHO it would be best to find a way to build a single arch agnostic base mirror image with a multiarch mirrror in it, but for now I think per arch version files are the minimum we need.